### PR TITLE
Added notifications for success/failure for offline operations

### PR DESCRIPTION
**Pull request by jhellar**
### Description

https://issues.jboss.org/browse/AEROGEAR-8345

### CODE REVIEW

**Comment by wtrocki** [comment]
Reminder to myself to add this option to ionic showcase app (for verification)

File: packages/sync/src/links/OfflineQueueLink.ts [hunk 1]
```diff
@@ -121,10 +120,16 @@ export class OfflineQueueLink extends ApolloLink {
     return new Promise((resolve, reject) => {
       opEntry.subscription = forward(operation).subscribe({
         next: result => {
+          if (this.listener && this.listener.onOperationSuccess) {
```
Comment (wtrocki) [inline packages/sync/src/links/OfflineQueueLink.ts]: [Oftopic] Optional chaining in typescript will transpile that. More info https://medium.com/inside-rimeto/optional-chaining-in-typescript-622c3121f99b

File: packages/sync/src/links/OfflineQueueLink.ts [hunk 2]
```diff
@@ -71,8 +71,7 @@ export class OfflineQueueLink extends ApolloLink {
       let result;
       try {
         result = await this.forwardQueuedOperation(opEntry);
-      } catch (error) {
-        // TODO: notify about failed operation via OfflineQueueListener
+      } catch (_) {
```
[the remark this answers is not in the document]
Comment (jhellar) [reply packages/sync/src/links/OfflineQueueLink.ts]: fixed

**Comment by jhellar** [comment]
@wtrocki thanks for review, I've added description to readme

**Pull request merged**